app/tests/test_item_to_collection_propagation.py

```diff
@@ -0,0 +1,115 @@
+from django.contrib.gis.geos import Polygon
+from django.test import TestCase
+
+from stac_api.models import Item
+from stac_api.utils import utc_aware
+
+import tests.database as db
+
+logger = logging.getLogger(__name__)
+
+
+class ItemsToCollectionTestCase(TestCase):
+    '''
+    Testing the propagation of item geometries to the bbox of the collection
+    '''
+    def setUp(self):
+        self.collection = db.create_collection('collection-1')
+        self.item = Item.objects.create(
+            collection=self.collection,
+            properties_datetime=utc_aware(datetime.utcnow()),
+            item_name='base-bbox',
+            geometry=GEOSGeometry(
+                'SRID=4326;POLYGON '
+                '((0 0, 0 45, 45 45, 45 0, 0 0))'
+            )
+        )
+        self.item.full_clean()
+        self.item.save()
+
+    def test_if_collection_gets_right_extent(self):
+        # the collection has to have the bbox of the item
+        self.assertEqual(self.collection.extent_geometry, self.item.geometry)
+
+    def test_if_new_collection_has_extent(self):
+        # a new collection as no bbox yet
+        collection_no_bbox = db.create_collection('collection-no-bbox')
+        self.assertIsNone(collection_no_bbox.extent_geometry)
+
+    def test_bigger_item(self):
+        # changing the size of the bbox of the collection
+        self.assertEqual(self.collection.extent_geometry, self.item.geometry)
+
+        bigger_item = Item.objects.create(
+            collection=self.collection,
+            properties_datetime=utc_aware(datetime.utcnow()),
+            item_name='bigger-bbox',
+            geometry=GEOSGeometry(
+                'SRID=4326;POLYGON '
+                '((0 0, 0 50, 50 50, 50 0, 0 0))'
+            )
+        )
+        bigger_item.full_clean()
+        bigger_item.save()
+        # collection has to have the size of the bigger extent
+        self.assertEqual(self.collection.extent_geometry, bigger_item.geometry)
+
+        bigger_item.delete()
+        self.assertEqual(self.collection.extent_geometry, self.item.geometry)
+
+    def test_smaller_item(self):
+        # changing the size of the bbox of the collection
+        self.assertEqual(self.collection.extent_geometry, self.item.geometry)
+        smaller_item = Item.objects.create(
+            collection=self.collection,
+            properties_datetime=utc_aware(datetime.utcnow()),
+            item_name='bigger-bbox',
+            geometry=GEOSGeometry(
+                'SRID=4326;POLYGON '
+                '((1 1, 1 40, 40 40, 40 1, 1 1))'
+            )
+        )
+        smaller_item.full_clean()
+        smaller_item.save()
+        # collection has to have the size of the bigger extent
+        self.assertEqual(self.collection.extent_geometry, self.item.geometry)
+        smaller_item.delete()
+        self.assertEqual(self.collection.extent_geometry, self.item.geometry)
+
+    def test_diagonal_item_update(self):
```

Comment:
Maybe test name could be a bit more explicit: `test_changing_bbox_with_diagonal_update`